apps/src/gamelab/animationListModule.js

```diff
@@ -164,6 +164,54 @@ function animationPropsReducer(state, action) {
 }
 
 /**
+ * Used to order numbers for sort()
+ */
+function compareNumbers(a, b) {
+  return a - b;
+}
+
+/**
+ * Given a name and an animationList, determine if the name is unique.
+ */
+export function isNameUnique(name, animationList) {
+  for (let animation in animationList) {
+    if (animationList[animation].name === name) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+ * Given a baseName and a animationList, provide a unique name
+ */
+function generateAnimationName(baseName, animationList) {
+  let unavailableNumbers = [];
+  for (let animation in animationList) {
+    let existingName = animationList[animation].name;
+    let lastUnderscore = existingName.lastIndexOf("_");
+    let lastDigits = existingName.substring(lastUnderscore + 1);
+    let existingBaseName = existingName;
+    if (/^\+?(0|[1-9]\d*)$/.test(lastDigits)) {
+      existingBaseName = existingName.substring(0, lastUnderscore);
+      if (existingBaseName === baseName) {
+        unavailableNumbers.push(parseInt(lastDigits));
```

Comment:
We've got a habit of commenting our regexes (especially anything semi-complicated) because not everyone on the team is super-comfortable with them.  That said, I'd go for it, they're really powerful.  In fact I bet you could do more with regex here.

Also, comments for your parameters might be good for this function - it's not immediately obvious what `baseName` is.

Here's what I would write.

```javascript
const re = new RegExp(`^${baseName}_(\\d+)$`)
let unavailableNumbers = animationList.reduce((unavailableNumbers, nextAnimation) => {
  let match = re.exec(nextAnimation.name);
  if (match !== null) {
    unavailableNumbers.push(parseInt(match[1]));
  }
  return unavailableNumbers
}, []);
```